core/sr-arithmetic/src/traits.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Primitives for the runtime modules.
+
+use rstd::prelude::*;
+use rstd::{self, convert::{TryFrom, TryInto}};
+//use crate::codec::{Codec, Encode, Decode, HasCompact};
+//use crate::generic::{Digest, DigestItem};
+//use crate::weights::DispatchInfo;
+use codec::HasCompact;
+pub use integer_sqrt::IntegerSquareRoot;
+pub use num_traits::{
+	Zero, One, Bounded, CheckedAdd, CheckedSub, CheckedMul, CheckedDiv,
+	CheckedShl, CheckedShr
+};
+use rstd::ops::{
+	Add, Sub, Mul, Div, Rem, AddAssign, SubAssign, MulAssign, DivAssign,
+	RemAssign, Shl, Shr
+};
+
+/// A lazy value.
+pub trait Lazy<T: ?Sized> {
```

Comment:
Not sure why these are here.

Reply:
I think that Lazy and Convert should stay in sr-primitive as they are not arithmetic related